src/lib/style-defaults.html

```diff
@@ -47,7 +50,25 @@
         return this._styleProperties;
       },
 
-      updateStyles: function() {
+      /**
+       * Re-evaluates and applies custom CSS properties to all elements in the
+       * document based on dynamic changes, such as adding or removing classes.
+       *
+       * For performance reasons, Polymer's custom CSS property shim relies
+       * on this explicit signal from the user to indicate when changes have
+       * been made that affect the values of custom properties.
+       *
+       * @method updateStyles
+       * @param {Object=} properties Properties object which is mixed into 
+       * the document root `customStyle` property. This argument provides a 
+       * shortcut for setting `customStyle` and then calling `updateStyles`.
+      */
+      updateStyles: function(properties) {
+        // force properties update.
+        this._properties = null;
+        if (properties) {
+          Polymer.Base.mixin(this.customStyle, properties);
+        }
         // invalidate the cache
         this._styleCache.clear();

```

Comment:
Users have reported on https://github.com/Polymer/polymer/issues/1851#issuecomment-112220994 that `Uncaught TypeError: Cannot read property 'clear' of undefined` can be thrown if `updateStyles` is called too early.  Should this just be guarded on `this._styleCache` being defined?  Or is this handled elsewhere?